sqlglot/dialects/databricks.py

```diff
@@ -51,6 +51,15 @@ class Generator(Spark.Generator):
             exp.ToChar: lambda self, e: self.function_fallback_sql(e),
         }
 
+        def columndef_sql(self, expression: exp.ColumnDef, sep: str = " ") -> str:
+            constraint = expression.find(exp.GeneratedAsIdentityColumnConstraint)
+            kind = self.sql(expression, "kind")
+            if constraint and isinstance(expression.this, exp.Identifier) and kind == "INT":
```

Comment:
What's the purpose of the instance check here? Also, the question remains: what if the column definition has a different integer type? For example, if you were transpiling a `SMALLINT` column from T-SQL to Databricks, then you wouldn't convert that into a `BIGINT` because you'd miss this branch. I think the correct approach is to get `kind = expression.args.get("kind")` and if it's a `DataType` then check if `kind.this` is in `DataType.INTEGER_TYPES`.

Another question here is what happens when you have integer types w/ length associated, e.g. `BIGINT(<num>)`. Right now we drop `<num>`, but should we perhaps keep it?

Reply:
- Spark doesn't support `INT(<num>)`, `BIGINT(<num>)` so I don't think it helps to preserve `<num>`, dropping <num> is appropriate.
- Thanks for the pointer to `DataTypeINTEGER_TYPES`